src/locutus/model/user_input.py

```diff
@@ -127,17 +133,21 @@ def create_or_replace_user_input(self, resource_type, collection_type, id, code,
         # Prep any existing data
         try:
             doc_ref = persistence().collection(resource_type).document(id) \
-                .collection(collection_type).document(code)
+                .collection(collection_type).document(mapped_pair)
 
             # Fetch existing data for the document if it exists
             doc_snapshot = doc_ref.get()
             existing_data = doc_snapshot.to_dict() if doc_snapshot.exists else {}
 
+            # Required for indexing
+            existing_data['code'] = code
+            existing_data['mapped_code'] = mapped_code 
+
              # Initialize type structure using return_format if it doesn't exist
             if type not in existing_data:
-                existing_data[type] = self.return_format() \
-                                      if callable(self.return_format) \
-                                      else self.return_format
+                existing_data[type] = user_input_instance.return_format() \
+                                    if callable(user_input_instance.return_format) \
+                                    else user_input_instance.return_format
```

Comment:
I think if you have to test an attribute as being callable or not probably indicates a design issue. I'm happy to meet to discuss ways to make this a much simpler return.